ConnectApiHelper.cls

```diff
@@ -143,21 +143,23 @@ global class ConnectApiHelper {
         List<ConnectApi.MessageSegmentInput> messageSegmentInputs = new List<ConnectApi.MessageSegmentInput>();
         Integer strPos = 0;
         // The pattern for matching mentions, markup begin/end tags, and inline images.
-        // The first group matches a 15 or 18 character ID surrounded by {}: 
+        // The first group matches a 15 or 18 character ID surrounded by {}:
         //   (\\{[a-zA-Z0-9]{15}\\}|\\{[a-zA-Z0-9]{18}\\})
         // The second/third groups match beginning/ending HTML tags: (<[a-zA-Z]*>)|(</[a-zA-Z]*>)
-        // The fourth group matches a 15 or 18 character content document ID preceded by "img:", 
-        //    optionally followed by a string (not containing '}'), and surrounded by {}: 
+        // The fourth group matches a 15 or 18 character content document ID preceded by "img:",
+        //    optionally followed by a string (not containing '}'), and surrounded by {}:
         //    (\\{img:(069[a-zA-Z0-9]{12,15})(:[\\s\\S]*?)?\\})
-        Pattern globalPattern = Pattern.compile('(\\{[a-zA-Z0-9]{15}\\}|\\{[a-zA-Z0-9]{18}\\})|(<[a-zA-Z]*>)|(</[a-zA-Z]*>)|(\\{img:(069[a-zA-Z0-9]{12,15})(:[\\s\\S]*?)?\\})');
+        // The fifth group matches a 12 or 18 character record ID preceded by "record:" ex:{record:01t3E000002GCm9QAG}
+        Pattern globalPattern = Pattern.compile('(\\{[a-zA-Z0-9]{15}\\}|\\{[a-zA-Z0-9]{18}\\})|(<[a-zA-Z]*>)|(</[a-zA-Z]*>)|(\\{img:(069[a-zA-Z0-9]{12,15})(:[\\s\\S]*?)?\\})|(\\{record:([a-zA-Z0-9]){12,18}(:[\\s\\S]*?)?\\})');
```

Comment:
Why {12,18} for "record:"? The ids are usually 15 or 18 chars long.

Reply:
Good catch -- I think it should be `{15,18}`.

Reply:
done + comment changed :)